### PR TITLE
INT-4459: Mention dependencies in the Docs

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4459

### CODE REVIEW

**Comment by artembilan** [comment]
@garyrussell ,

let me know if you are OK with such a solution.
I'm going to add similar block with an explicit mentioning of the SI artifact in each chapter.
Whenever it is necessary, I will mention some other required or optional artifacts.

So, WDYT?

**Comment by garyrussell** [comment]
Hmmm.... perhaps another table like this one https://docs.spring.io/spring-integration/reference/html/endpoint-summary.html would be better than having it spread all over the docs.

But I guess either way is fine.

**Comment by artembilan** [comment]
> perhaps another table like this one

Hm. I would say that it sounds like another column for that table, but I guess it is going to uglier...

I think everybody will find an explicit mentioning in the beginning of chapter as convenient.
Also I think we need to add something like `Maven/Gradle` toggle (note sure how to do this yet...). Plus include a current version.
And everything that for easier copy-n-paste experience.

**Comment by garyrussell** [comment]
ok.

**Comment by garyrussell** [comment]
But maybe reference that when using the boot starter, the version is managed by boot and is not needed (and should not be provided).

**Comment by artembilan** [comment]
Ready for review.

**Comment by garyrussell** [comment]
```
> Task :asciidoctor
Converting /Users/grussell/Development/spring-integration/src/reference/asciidoc/index.adoc
asciidoctor: WARNING: spel.adoc: line 40: invalid style for paragraph: appendix
asciidoctor: WARNING: message-publishing.adoc: line 37: invalid style for paragraph: appendix
asciidoctor: WARNING: transactions.adoc: line 81: invalid style for paragraph: appendix
asciidoctor: WARNING: security.adoc: line 17: invalid style for paragraph: appendix
asciidoctor: WARNING: configuration.adoc: line 24: invalid style for paragraph: appendix
asciidoctor: WARNING: testing.adoc: line 64: invalid style for paragraph: appendix
asciidoctor: WARNING: samples.adoc: line 44: invalid style for paragraph: appendix
asciidoctor: WARNING: changes-2.0-2.1.adoc: line 163: invalid style for paragraph: appendix
asciidoctor: WARNING: changes-2.0-2.1.adoc: line 170: invalid style for paragraph: appendix
asciidoctor: WARNING: /Users/grussell/Development/spring-integration/src/reference/asciidoc/index.adoc: stylesheet does not exist or cannot be read: /Users/grussell/Development/spring-integration/src/reference/asciidoc/golo.css
Converting /Users/grussell/Development/spring-integration/src/reference/asciidoc/index.adoc
asciidoctor: WARNING: spel.adoc: line 40: invalid style for paragraph: appendix
asciidoctor: WARNING: message-publishing.adoc: line 37: invalid style for paragraph: appendix
asciidoctor: WARNING: transactions.adoc: line 81: invalid style for paragraph: appendix
asciidoctor: WARNING: security.adoc: line 17: invalid style for paragraph: appendix
asciidoctor: WARNING: configuration.adoc: line 24: invalid style for paragraph: appendix
asciidoctor: WARNING: testing.adoc: line 64: invalid style for paragraph: appendix
asciidoctor: WARNING: samples.adoc: line 44: invalid style for paragraph: appendix
asciidoctor: WARNING: changes-2.0-2.1.adoc: line 163: invalid style for paragraph: appendix
asciidoctor: WARNING: changes-2.0-2.1.adoc: line 170: invalid style for paragraph: appendix
<==-----------> 16% EXECUTING [25s]
```

I've seen that before with mis-matched `====`.

File: src/reference/asciidoc/endpoint-summary.adoc [hunk 1]
```diff
@@ -3,6 +3,24 @@
...
+            <groupId>org.springframework.integration</groupId>
+            <artifactId>spring-integration-bom</artifactId>
+            <version>{project-version}</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+----
+====
```
Comment (garyrussell) [inline src/reference/asciidoc/endpoint-summary.adoc]: Missing before.

Comment (artembilan) [reply src/reference/asciidoc/endpoint-summary.adoc]: Fixed

**Comment by garyrussell** [comment]
Hmmm - doesn't seem to be related to this PR, but I see

```
asciidoctor: WARNING: /Users/grussell/Development/spring-integration/src/reference/asciidoc/index.adoc: stylesheet does not exist or cannot be read: /Users/grussell/Development/spring-integration/src/reference/asciidoc/golo.css
```

**Comment by garyrussell** [comment]
Looks like the asciidoctor plugin change from 1.5.3 to 1.5.8 introduced it.

**Comment by artembilan** [comment]
Right, I've seen that as well, but have no idea how to fix: not too much clear docs about Gradle plugin for Asciidoc.
Or this might be a problem of the Docbook plugin...

We can consider to revert to the mentioned version or some intermediate which works fine for us.

On the other hand I don't see any problems with our target doc.

Do you know what and how to look to determine that we are missing anything?

**Pull request merged**